File: food-donation-backend/src/main/java/com/fooddonation/service/UserService.java
package com.fooddonation.service;

import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import com.fooddonation.dto.response.PageResponse;
import com.fooddonation.dto.response.UserResponse;
import com.fooddonation.enums.UserStatus;

public interface UserService {

	UserResponse getCurrentUser(String email);

	UserResponse getUserById(Long id);

	UserResponse updateProfile(String email, Object updateRequest);

	String uploadProfileImage(String email, MultipartFile file);

	void changePassword(String email, String oldPassword, String newPassword);

	PageResponse<UserResponse> getAllUsers(String role, String query, Pageable pageable);

	void updateUserStatus(Long userId, UserStatus status);

	void deleteUser(Long userId);

	UserResponse approveNgo(Long userId);

	UserResponse approveDonor(Long userId);

}

File: food-donation-frontend/src/services/userService.js
import apiClient from './apiClient';

const userService = {
  async getCurrentUser() {
    const { data } = await apiClient.get('/users/me');
    return data.data;
  },
  async getUserById(id) {
    const { data } = await apiClient.get(`/users/${id}`);
    return data.data;
  },
  async uploadProfileImage(file) {
    const formData = new FormData();
    formData.append('file', file);
    const { data } = await apiClient.post('/users/me/profile-image', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return data.data;
  },
  async changePassword(oldPassword, newPassword) {
    const { data } = await apiClient.put('/users/me/change-password', { oldPassword, newPassword });
    return data;
  },
};

export default userService;
